pydantic_settings/sources.py

```diff
@@ -126,6 +127,28 @@ class PydanticBaseSettingsSource(ABC):
     def __init__(self, settings_cls: type[BaseSettings]):
         self.settings_cls = settings_cls
         self.config = settings_cls.model_config
+        self._previous_states: dict[str, dict[str, Any]] = {}
+
+    def _set_previous_states(self, states: dict[str, dict[str, Any]]) -> None:
+        """
+        Record the state of settings from all previous settings sources. This should
+        be called right before __call__.
+        """
+        self._previous_states = states
+
+    @property
+    def previous_states(self) -> dict[str, dict[str, Any]]:
+        """
+        The state of all previous settings sources.
+        """
+        return self._previous_states
+
+    @property
+    def previous_state(self) -> dict[str, Any]:
```

Comment:
Do we really need this property here?

Reply:
In my opinion, this property covers the majority of use-cases and avoids end-users the trouble of having to scan through all entries of the `previous_states` dictionary to find the key they are looking for.

As an example, I have opened this PR because I would like to read a "secret name" from a previous settings source and use it to retrieve a JSON document from AWS Secrets Manager. It does not matter to me whether the secret name comes from env vars, a dotenv file or the cli arguments. Therefore, the `previous_state` property is the one I would like to use.

Reply:
Ok, that makes sense.

Reply:
Ok, that makes sense. As we have the `state` in `main.py` loop, isn't it better to set the state there? I mean we already have `deep_update` there.

Reply:
I'm not sure what you mean. Something like [this diff](https://topaz.github.io/paste/#XQAAAQB+BwAAAAAAAAAyGkkJwvwRfVDeCk9H2J4aD4LFWYdfouH8Zl+cjlbJpqoN1QDlrZOpnDV0gB3D+Kwc3qYU+JNV10stFOHO+K0AXsQWyEXAZyffe+a9Gw2xJBVc15TqbudD7c8WIHu6zatZsXk26uI4+aNT3bEyfs4KSOTjXXWY+HdcMrUhWULnr8i2DM+yhJkTCtJwkBoPkvUaKmzi+PMEbP6j+hl/Hp3iEl227e8dgaMJ0uycxTc5J/N2LRGH2aV2dBTBJAnho0zvJ2JMt5f1OQtgI3g5O+yG1nYWtox6rC9Jkmf5g4t/wXOxqRwemC42PBuKRnP1YMnB26eRkrUzK1JXl1jkooPyE0C7TlPrPBzPgdM57zWGXK//pa4fc9FHmRaVesjnOEZRSi9k7MXGuyVCTZ7dn5guQkJijTkuSV8U51ivqVASps/f2nrXNMCynPcrR99VoOr8+b0y0zI7U5Nd66YvON10NYEFBHN8xMqYovPS5GxToHXIc223kgVhNtKXaSOdmCbLo5MYvTbUAL6DtI9X5RM5D/RNEbGvjcx1tVVRf6H8lPCC+rDFB/0OGqT+QbyLVuMGm1uQgE/EKkliZjvr+1ZikDAJdDZjYjo9YQgIZF203WKbzuU4rWtTZXPGmlh9Brx+s6xrQcY3Wy2cmyu/Qg85nF0VpudqDZ6eg6uRrEer181PbNyLo3YsABrxklznUNyGuSStCU+c3B1nSP3z6W6/KdYIaSZG9kRkIHdz7JuQHuPUXnH7u+XbLixOl8rA9vvlwhWAMMdKrLmYEHnb74cyYfoJyzQaO6BQDQ31KRCyYQ9iTpWlJg3p26BNn4ronjAQvgO5iDx/emq4g/A8lTKr3vv3sK+J5DCE+u1Ewt7t3Pgz/0thXWR4OkHvZ76HYP/ofxek)?

Reply:
Exactly!

Reply:
Done!

Reply:
Thanks!